### PR TITLE
fix(SITES-25655): [Regression] Table _setHeaderCellScope should not update headerCell.content before it exists

### PR DESCRIPTION
**Pull request by majornista**
## Description
Fixes race condition where `_setHeaderCellScope` is called before `headerCell.content` has initialized.

## Related Issue
SITES-25655

## Motivation and Context
Fixes regression where the `_setHeaderCellScope` gets called before `headerCell.content` has initialized:
https://github.com/adobe/coral-spectrum/blob/369399a1dcaf0f0883037da0efd86039c2b40ae0/coral-component-table/src/scripts/Table.js#L2749-L2751

Following https://github.com/adobe/coral-spectrum/pull/346 and its merge into QS in CQ/quickstart/pull/37653, the Rollout Page button in the editors have stopped working, while displaying the following errors in the console:
```
Uncaught TypeError: Cannot read properties of null (reading 'setAttribute')
    at HTMLTableElement.value (coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:28:769217)
    at coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:28:770885
    at Array.forEach (<anonymous>)
    at coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:28:770862
    at Array.forEach (<anonymous>)
    at HTMLTableElement.value (coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:28:770835)
    at HTMLTableElement.value (coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:4:49176)
    at HTMLTableElement.value (coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:4:55052)
    at l.connectedCallback (coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:2:34083)
    at d (coralui3.lc-e611376559522fd7d3d8ceef633171b0-lc.min.js:2:32155)
```

One workaround would be to explicitly use `coral-table-headercell-content` to wrap the table cell content within the template that defines the `coral-table`, but this would be problematic for backwards compatibility, since it is sort of an implementation detail, that is not publicly documented: https://developer.adobe.com/experience-manager/reference-materials/6-5/coral-ui/coralui3/Coral.Table.html#Coral.Table.HeaderCell:api.
 
## How Has This Been Tested?
Tested following steps described in SITES-25655

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by mariang-ens — APPROVED** [review]

**Comment by majornista** [comment]
@mariang-ens Do I need a second review, or can this be merged?

**Comment by mariang-ens** [comment]
@majornista I think it's okay to merge this PR

**Pull request merged**
